src/Storage/Scripting/Evaluator.cs

```diff
@@ -112,7 +89,34 @@ private string ReplaceTags(Match match)
                 }
                 return msgBuilder.ToString();
             }
+
             return result;
         }
+        internal static string GetFullTagName(string tagName)
+        {
+            return $"evaluator-{tagName}";
+        }
+        internal static string GetFullTagName(Type evaluatorType)
+        {
+            if (evaluatorType?.GetCustomAttributes(typeof(ScriptTagNameAttribute), false).FirstOrDefault() is ScriptTagNameAttribute tagAttribute)
+                return GetFullTagName(tagAttribute.TagName);
+
+            return string.Empty;
+        }
+    }
+
+    public static class EvaluatorExtensions
+    {
+        public static IRepositoryBuilder UseEvaluator(this IRepositoryBuilder repositoryBuilder, IEvaluator evaluator)

```

Comment:
My suggestion: UseScriptEvaluator can be better because many features can use the "evaluator" word.